### DISCUSSION TITLE
Compose templates

### DISCUSSION THREAD

**Discussion by pylang** [copier-org/copier#373] [General]
I believe I heard on the [podcast](https://www.pythonpodcast.com/copier-project-scaffolding-episode-297/)  (~19m) that we can use templates to make other templates.   I haven't seen this feature explained the docs.  How does this work?  Are there example templates?

It may be worth mentioning this the [features list in the docs](https://copier.readthedocs.io/en/stable/comparisons/).

Example:

```bash
   📁   temp_gitignore/
   📄   ├── .gitignore
   📄   └── copier-answers.yml

   📁   temp_pyproject/
   📄   ├── .pyproject.toml
   📄   └── copier-answers.yml

   📁   temp_gitignore_and_pyproject/
   📄   ├── ...?
   📄   └── copier-answers.yml ?
```

**Comment by yajo** [comment]
Yes, that's a fairly new avenue for templates. It's not documented yet, and you can subscribe to #286 to track that documentation progress.

The short answer is that you need to [make use of `_subdirectory` in the template](https://copier.readthedocs.io/en/stable/configuring/#subdirectory). One example is https://github.com/copier-org/autopretty, which is a template that can be applied to both normal projects and templates, and that is applied to itself.
